### PR TITLE
Re-use CMFS nodes that need to be placed for a namespaces

### PR DESCRIPTION
**Pull request by kwiatekus**
**Related issue(s)**
https://github.com/kyma-project/console/issues/1153

### CODE REVIEW

File: core/src/luigi-config/luigi-config.js [hunk 1]
```diff
@@ -601,6 +598,16 @@ Promise.all(initPromises)
                 }
                 return [];
               });
+          window.clusterMicrofrontendNodesForNamespace =
+              cmfs
+                .filter(cmf => cmf.placement === 'namespace')
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]: Please note that after environment -> namespace renaming we kept the possibility to set the placement also to `environment` to be backwards compatible. In previous approach I see that it was accidentally changed (it should be `getUiEntities('clustermicrofrontends', namespace, [ 'namespace',	'environment'])`) and probably we should fix it here ;) 
```suggestion
                .filter(cmf => cmf.placement === 'namespace' || cmf.placement === 'environment')
```

File: core/src/luigi-config/luigi-config.js [hunk 2]
```diff
@@ -228,10 +228,7 @@ function getNodes(context) {
   ];
   return Promise.all([
     getUiEntities('microfrontends', namespace),
-    getUiEntities('clustermicrofrontends', namespace, [
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]: Please cleanup the `getUiEntities` function. We don't need to worry about CMFs there anymore so we should simplify it

Comment (kwiatekus) [reply core/src/luigi-config/luigi-config.js]: done

File: core/src/luigi-config/luigi-config.js [hunk 3]
```diff
@@ -277,52 +274,59 @@ async function getNamespace(namespaceName) {
...
+    }
+  }`;
+
   return (
     fromCache ||
-    fetchFromKyma(fetchUrl)
+    fetchFromGraphQL(query, {namespace}, true)
       .then(result => {
-        if (!result.items.length) {
+        if (!result.microFrontends.length) {
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]:
```suggestion
        if (!result.microFrontends || !result.microFrontends.length) {
```

File: core/src/luigi-config/luigi-config.js [hunk 4]
```diff
@@ -277,52 +274,59 @@ async function getNamespace(namespaceName) {
...
-            // placement only exists in clustermicrofrontends
-            return !placements || placements.includes(item.spec.placement);
-          })
+        return result.microFrontends
           .map(function (item) {
-
-            if (item.spec.navigationNodes) {
-              var tree = convertToNavigationTree(item.metadata.name, item.spec, config, navigation, consoleViewGroupName, segmentPrefix);
+            if (item.navigationNodes) {
+              var tree = convertToNavigationTree(item.name, item, config, navigation, consoleViewGroupName, segmentPrefix);
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]:
```suggestion
              const tree = convertToNavigationTree(item.name, item, config, navigation, consoleViewGroupName, segmentPrefix);
```

File: core/src/luigi-config/luigi-config.js [hunk 5]
```diff
@@ -277,52 +274,59 @@ async function getNamespace(namespaceName) {
...
               return tree;
             }
             return [];
           });
       })
       .catch(err => {
-        console.error('Error fetching UiEntity ' + name, err);
+        console.error('Error fetching Microfrontend ' + name, err);
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]:
```suggestion
        console.error(`Error fetching Microfrontend ${name}: ${err}`);
```

File: core/src/luigi-config/luigi-config.js [hunk 6]
```diff
@@ -601,6 +605,16 @@ Promise.all(initPromises)
                 }
                 return [];
               });
+          window.clusterMicrofrontendNodesForNamespace =
+              cmfs
+                .filter(cmf => cmf.placement === 'namespace' || cmf.placement === 'environment')
+                .map(cmf => {
+                  if (cmf.navigationNodes) {
+                    var tree = convertToNavigationTree(cmf.name, cmf, config, navigation, consoleViewGroupName, 'cmf-');
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]:
```suggestion
                    const tree = convertToNavigationTree(cmf.name, cmf, config, navigation, consoleViewGroupName, 'cmf-');
```

File: core/src/luigi-config/luigi-config.js [hunk 7]
```diff
@@ -277,52 +274,59 @@ async function getNamespace(namespaceName) {
...
-  if (namespace) {
-    const currentNamespace = await getNamespace(namespace);
-  }
-  var fetchUrl =
-    k8sServerUrl +
-    '/apis/ui.kyma-project.io/v1alpha1/' +
-    (namespace ? 'namespaces/' + namespace + '/' : '') +
-    entityname;
-  const segmentPrefix = entityname === 'clustermicrofrontends' ? 'cmf-' : 'mf-';
+async function getMicrofrontends(namespace) {
```
Comment (dariadomagala) [inline core/src/luigi-config/luigi-config.js]:
```suggestion
const getMicrofrontends = async (namespace) => {
```

**Pull request merged**